File: trunk/external/zer0/src/RGSS/Tone.cpp
#include <ruby.h>

#include <hltypes/util.h>

#include "RGSS/Tone.h"

namespace zer0
{
	namespace RGSS
	{
		VALUE rb_cTone;

		void Tone::createRubyInterface()
		{
		}

		Tone::Tone()
		{
			this->set(255.0f, 255.0f, 255.0f);
		}

		Tone::Tone(float r, float g, float b, float gr)
		{
			this->set(r, g, b, gr);
		}

		Tone::~Tone()
		{

		}

		void Tone::setRed(float value)
		{
			this->red = hclamp(value, -255.0f, 255.0f);
		}

		void Tone::setGreen(float value)
		{
			this->green = hclamp(value, -255.0f, 255.0f);
		}

		void Tone::setBlue(float value)
		{
			this->blue = hclamp(value, -255.0f, 255.0f);
		}

		void Tone::setGray(float value)
		{
			this->gray = hclamp(value, -255.0f, 255.0f);
		}

		void Tone::set(float r, float g, float b, float gr)
		{
			this->red = hclamp(r, -255.0f, 255.0f);
			this->green = hclamp(g, -255.0f, 255.0f);
			this->blue = hclamp(b, -255.0f, 255.0f);
			this->gray = hclamp(gr, -255.0f, 255.0f);
		}
	}
}

File: trunk/editor/ARCed.NET/ARCDumpTests/RPG/Tone.cs

// TODO: Clamp values

class Tone
{
	public float red { get; set; }
	public float green { get; set; }
	public float blue { get; set; }
	public float gray { get; set; }

	public Tone() : this(0, 0, 0, 0) { }

	public Tone(float red, float green, float blue) :
		this(red, green, blue, 255) { }

	public Tone(float red, float green, float blue, float gray)
	{
		this.red = red;
		this.green = green;
		this.blue = blue;
		this.gray = gray;
	}

	public void set(float red, float green, float blue)
	{
		set(red, green, blue, this.gray);
	}

	public void set(float red, float green, float blue, float gray)
	{
		this.red = red;
		this.green = green;
		this.blue = blue;
		this.gray = gray;
	}
}
